main.py

```diff
@@ -13,171 +14,172 @@
-	Args:
-		frame:
-		faces:
-	Returns:
-	face drawn processed frame
-	"""
-	for (x, y, w, h) in faces:
-		xA = x
-		yA = y
-		xB = x + w
-		yB = y + h
-		cv2.rectangle(frame, (xA, yA), (xB, yB), (0, 255, 0), 2)
-	return frame
+    """
+    draw rectangle around detected faces
+    Args:
+        frame:
+        faces:
+    Returns:
+    face drawn processed frame
+    """
+    for (x, y, w, h) in faces:
+        xA = x
+        yA = y
+        xB = x + w
+        yB = y + h
+        cv2.rectangle(frame, (xA, yA), (xB, yB), (0, 255, 0), 2)
+    return frame
 
 
 def put_label_on_face(frame, faces, labels):
-	"""
-	draw label on faces
-	Args:
-		frame:
-		faces:
-		labels:
-	Returns:
-		processed frame
-	"""
-	i = 0
-	for x, y, w, h in faces:
-		cv2.putText(frame, str(labels[i]), (x, y), font, 1, (255, 255, 255), 2)
-		i += 1
-	return frame
+    """
+    draw label on faces
+    Args:
+        frame:
+        faces:
+        labels:
+    Returns:
+        processed frame
+    """
+    i = 0
+    for x, y, w, h in faces:
+        cv2.putText(frame, str(labels[i]), (x, y), font, 1, (255, 255, 255), 2)
+        i += 1
+    return frame
+
 
 def background_subtraction(previous_frame, frame_resized_grayscale, min_area):
-	"""
-	This function returns 1 for the frames in which the area 
-	after subtraction with previous frame is greater than minimum area
-	defined. 
-	Thus expensive computation of human detection face detection 
-	and face recognition is not done on all the frames.
-	Only the frames undergoing significant amount of change (which is controlled min_area)
-	are processed for detection and recognition.
-	"""
-	frameDelta = cv2.absdiff(previous_frame, frame_resized_grayscale)
-	thresh = cv2.threshold(frameDelta, 25, 255, cv2.THRESH_BINARY)[1]
-	thresh = cv2.dilate(thresh, None, iterations=2)
-	im2, cnts, hierarchy = cv2.findContours(thresh.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
-	temp=0
-	for c in cnts:
-		# if the contour is too small, ignore it
-		if cv2.contourArea(c) > min_area:
-			temp=1
-	return temp		
+    """
+    This function returns 1 for the frames in which the area
+    after subtraction with previous frame is greater than minimum area
+    defined.
+    Thus expensive computation of human detection face detection
+    and face recognition is not done on all the frames.
+    Only the frames undergoing significant amount of change (which is controlled min_area)
+    are processed for detection and recognition.
+    """
+    frameDelta = cv2.absdiff(previous_frame, frame_resized_grayscale)
+    thresh = cv2.threshold(frameDelta, 25, 255, cv2.THRESH_BINARY)[1]
+    thresh = cv2.dilate(thresh, None, iterations=2)
+    im2, cnts, hierarchy = cv2.findContours(thresh.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+    temp = 0
+    for c in cnts:
+        # if the contour is too small, ignore it
+        if cv2.contourArea(c) > min_area:
+            temp = 1
+    return temp
+
 
 if __name__ == '__main__':
-	"""
```

Comment:
what's the improvement in this refactor?

Reply:
It's the standard IntelliJ/Pycharm refactor. Personally I think it's more readable like this, but the most important thing is the lbph creation which got changed in OpenCV3.4.

Also might be handy to set the standard path to the video directory in the argparse.